### DISCUSSION TITLE
[BUG] Failed to complete sh install-gui.sh on Amazon Linux 2

### DISCUSSION THREAD

**Discussion by franzqat** [Chia-Network/chia-blockchain#3439] [Support]
**Describe the bug**
After following the installation guide from [here](https://github.com/Chia-Network/chia-blockchain/wiki/INSTALL#amazon-linux-2) and tried to follow the [quick start guide](https://github.com/Chia-Network/chia-blockchain/wiki/Quick-Start-Guide#run-a-full-node--farmer--harvester--wallet) 
I got stuck at sh install-gui.sh point because of this error.

[2021-05-02T10_35_23_964Z-debug.log](https://github.com/Chia-Network/chia-blockchain/files/6411174/2021-05-02T10_35_23_964Z-debug.log)
```
0 info it worked if it ends with ok
1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'run', 'build' ]
2 info using npm@6.14.12
3 info using node@v12.22.1
4 verbose run-script [ 'prebuild', 'build', 'postbuild' ]
5 info lifecycle chia-blockchain@~prebuild: chia-blockchain@
6 info lifecycle chia-blockchain@~build: chia-blockchain@
7 verbose lifecycle chia-blockchain@~build: unsafe-perm in lifecycle true
8 verbose lifecycle chia-blockchain@~build: PATH: /usr/lib/node_modules/npm/node_modules/npm-lifecycle/node-gyp-bin:/home/ec2-user/chia-blockchain/chia-blockchain-gui/node_modules/.bin:/home/ec2-user/chia-blockchain/venv/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/home/ec2-user/.local/bin:/home/ec2-user/bin
9 verbose lifecycle chia-blockchain@~build: CWD: /home/ec2-user/chia-blockchain/chia-blockchain-gui
10 silly lifecycle chia-blockchain@~build: Args: [
10 silly lifecycle   '-c',
10 silly lifecycle   'npm run locale && npm run build:electron && npm run build:react'
10 silly lifecycle ]
11 silly lifecycle chia-blockchain@~build: Returned: code: 1  signal: null
12 info lifecycle chia-blockchain@~build: Failed to exec build script
13 verbose stack Error: chia-blockchain@ build: `npm run locale && npm run build:electron && npm run build:react`
13 verbose stack Exit status 1
13 verbose stack     at EventEmitter.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/index.js:332:16)
13 verbose stack     at EventEmitter.emit (events.js:314:20)
13 verbose stack     at ChildProcess.<anonymous> (/usr/lib/node_modules/npm/node_modules/npm-lifecycle/lib/spawn.js:55:14)
13 verbose stack     at ChildProcess.emit (events.js:314:20)
13 verbose stack     at maybeClose (internal/child_process.js:1022:16)
13 verbose stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:287:5)
14 verbose pkgid chia-blockchain@
15 verbose cwd /home/ec2-user/chia-blockchain/chia-blockchain-gui
16 verbose Linux 4.14.231-173.360.amzn2.x86_64
17 verbose argv "/usr/bin/node" "/usr/bin/npm" "run" "build"
18 verbose node v12.22.1
19 verbose npm  v6.14.12
20 error code ELIFECYCLE
21 error errno 1
22 error chia-blockchain@ build: `npm run locale && npm run build:electron && npm run build:react`
22 error Exit status 1
23 error Failed at the chia-blockchain@ build script.
23 error This is probably not a problem with npm. There is likely additional logging output above.
24 verbose exit [ 1, true ]

```

**To Reproduce**

1. Follow installation on Amazon Linux 2 on free tier t2.micro instance
2. Follow quick start guide until [here](https://github.com/Chia-Network/chia-blockchain/wiki/Quick-Start-Guide#run-a-full-node--farmer--harvester--wallet)
3. run sh install-gui.sh 

**Desktop (please complete the following information):**

AWS free t2.micro instance

**Comment by sargonas** [comment]
The GUI installer step is run under the assumption you’re on a graphical desktop mode. Remotely connecting to an AWS instance however does not provide a desktop, therefore this step will fail since no desktop renderers are available. If you followed the rest of the install steps up until this point, then you should be safe to move forward using the command line interface without the GUI and good to go.

**Comment by BrandtH22** [comment]
If you are still experiencing the issues it is recommended to upgrade to the latest version of chia. Please download it from here: https://www.chia.net/downloads/

Since this issue has been open for a number of years without additional comments, we will be closing this ticket but if we have closed this ticket in error do not hesitate to reach out to us again with any followup questions, comments, or if the issue persists after an update.

```
Please note that upgrading major versions of chia generally requires the config.yaml to be recreated using these steps:
1. Stop chia
2. Rename the config or move it to a different directory
3. In CLI run chia init
4. Copy any custom settings from the old config to the new (plot drive directories, pool info, trusted node info, etc) - be careful here as yaml is very picky about formatting including those leading spaces
5. Start Chia
```

The best place to reach our support team is on Discord (https://discord.gg/chia) or by reopening this ticket.
